image/tarexport/save.go

```diff
@@ -516,20 +522,21 @@ func (s *saveSession) saveLayer(id layer.ChainID, legacyImg image.V1Image, creat
 		}
 
 		s.diffIDPaths[l.DiffID()] = layerPath
-		s.savedLayers[legacyImg.ID] = struct{}{}
```

Comment:
We don't need `savedLayers` any more?

Reply:
Whoops, restored. I didn't intend to remove it in this PR (it _could_ be deleted though, but that's out of scope here).

Reply:
Hm, maybe we _do_ want to remove the saved layers, there is [this test](https://github.com/moby/moby/blob/33b5a5813b805aaba28c06ee1a8aa5b1f2435a1d/image/tarexport/save.go#L454) on top, don't we want to test `diffIDDescs` at the beginning and return that desc if it exists? Right now it will return an empty desc for already saved layers

Reply:
Removed unused maps and refactored the function to better distinguish between saved image (Config) and saved layer (diff tar archive).